io/io/src/TStreamerInfo.cxx

```diff
@@ -5125,6 +5135,19 @@ void TStreamerInfo::PrintValueSTL(const char *name, TVirtualCollectionProxy *con
    printf("\n");
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Replace the TClass this streamerInfo is pointing to (belongs to)
+
+void TStreamerInfo::SetClass(TClass *newcl)
+{
+   if (newcl) {
+      // This is mostly (but not only) for the artificial "This" streamerElement for an stl collection.
+      Update(fClass, newcl);
+      fClass = newcl;
+   } else
+      newcl = nullptr;
```

Comment:
This seems wrong? You possibly meant:
```suggestion
   if (newcl) {
      // This is mostly (but not only) for the artificial "This" streamerElement for an stl collection.
      Update(fClass, newcl);
   }
   fClass = newcl;
```

Note how that sets `fClass` also if `!newcl`.

Reply:
indeed.